chat-plugins/mafia.js

```diff
@@ -780,6 +807,165 @@ class MafiaTracker extends Rooms.RoomGame {
 			this.dead[i].destroy();
 		}
 	}
+
+	//IDEA commands
+	ideaInit(target) {
+		this.originalRoles = [];
+		this.originalRoleString = '';
+		this.roles = [];
+		this.roleString = '';
+
+		this.IDEA.data = MafiaData.IDEAs[target];
+		if (typeof this.IDEA.data === 'string') this.IDEA.data = MafiaData.IDEAs[this.IDEA.data];
+		if (typeof this.IDEA.data !== 'object') return false;
+		if (!this.IDEA.data) return `${target} is not a valid IDEA.`;
+		this.sendRoom(`${this.IDEA.data.name} roles loaded.`, {declare: true});
+		return this.ideaDistributeRoles(target);
+	}
+
+	ideaDistributeRoles(target) {
+		if (!this.IDEA.data) return `No IDEA module loaded`;
+		if (this.phase !== 'locked' && this.phase !== 'IDEAlocked') return `The game must be in a locked state to distribute IDEA roles.`;
+
+		const neededRoles = this.IDEA.data.choices * this.playerCount;
+		if (neededRoles > this.IDEA.data.roles.length) return `Not enough roles given`;
+
+		let roles = [];
+		let selectedIndexes = [];
+		for (let i = 0; i < neededRoles; i++) {
+			let randomIndex;
+			do {
+				randomIndex = Math.floor(Math.random() * this.IDEA.data.roles.length);
+			} while (selectedIndexes.indexOf(randomIndex) !== -1);
+			roles.push(this.IDEA.data.roles[randomIndex]);
+			selectedIndexes.push(randomIndex);
+		}
+		Dex.shuffle(roles);
+		this.IDEA.waitingPick = [];
+		for (const p in this.players) {
+			const player = this.players[p];
+			player.IDEA = {};
+			player.IDEA.choices = roles.splice(0, this.IDEA.data.choices);
+			player.IDEA.originalChoices = player.IDEA.choices.slice();
+			player.IDEA.picks = {};
+			for (const pick of this.IDEA.data.picks) {
+				player.IDEA.picks[pick] = null;
+				this.IDEA.waitingPick.push(p);
+			}
+		}
+		this.phase = 'IDEApicking';
+		this.updatePlayers();
+
+		this.sendRoom(`IDEA roles have been distributed. You will have ${IDEA_TIMER / 1000} seconds to make your picks.`, {declare: true});
+		this.IDEA.timer = setTimeout(() => { this.ideaFinalizePicks(); }, IDEA_TIMER);
+
+		return ``;
+	}
+
+	ideaPick(userid, target) {
+		let buf = '';
+		if (this.phase !== 'IDEApicking') return 'The game is not in the IDEA picking phase.';
+		const player = this.players[userid];
+		target = target.map((item) => toId(item));
```

Comment:
Mostly a style thing, but no parentheses are needed if the arrow function only has one argument.